modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -641,7 +641,26 @@ def to_pandas(cls, partitions):
         pandas.DataFrame
             A pandas DataFrame
         """
-        retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
+        retrieved_objects = cls.get_objects_from_partitions(partitions.flatten())
+        if all(
+            map(
+                lambda obj: isinstance(obj, (pandas.DataFrame, pandas.Series)),
+                retrieved_objects,
+            )
+        ):
+            height, width = 0, 0
+            if len(partitions.shape) == 2:
+                # possible case when shape == (0,)
+                height, width = partitions.shape
+            # restore 2d array
+            retrieved_objects = [
+                [retrieved_objects[i * width + j] for j in range(width)]
+                for i in range(height)
+            ]
```

Comment:
would it give the same result?
```suggestion
            retrieved_objects = retrieved_objects.reshape(partitions.shape)
```

Reply:
There was some problem with it, I don't remember exactly what. Let's try again.

Reply:
The problem is that `reshape` does not work as expected when the elements of the sequence are dataframes. For example, you can see this effect in the following command: `np.reshape([pd.DataFrame([1,2,3])], (1,1))`